### ISSUE TITLE
Activity causes inactive session to become active session in current client.

### ISSUE THREAD

**Issue by systemmonkey42** [tmux/tmux#1995]
### Issue description

Please read https://github.com/tmux/tmux/blob/master/.github/CONTRIBUTING.md
before opening an issue.

If you have upgraded, make sure your issue is not covered in the CHANGES file
for your version: https://raw.githubusercontent.com/tmux/tmux/master/CHANGES

Describe the problem and the steps to reproduce. Add a minimal tmux config if
necessary. Screenshots can be helpful, but no more than one or two.

Do not report bugs (crashes, incorrect behaviour) without reproducing on a tmux
built from the latest code in Git.

### Required information

Please provide the following information:

* tmux version (`tmux -V`).
  tmux next-3.1   < built from current HEAD (3bb11ec4)
* Platform (`uname -sp`).
  Linux x86_64
* $TERM inside and outside of tmux (`echo $TERM`).
   inside:  screen
   outside: xterm-256color
* Logs from tmux (`tmux kill-server; tmux -vv new`).
   Let me know if you need these.

For the purposes of testing, I remove .tmux.conf so all settings are default.

### Description

The active client automatically switches sessions based on activity.

If I have two sessions, one attached and one detached.  Any display activity causes it to become the active session in the last active client.

#### example
$ tmux new
<new session>
$ tmux new -d
<new session launches in background, but immediately becomes the active session due to the shell displaying the prompt>
$ sleep 5; ls 
<set up for delayed output>
Ctrl-B )   <switch to original session>
<5 seconds later, current client automatically switches back to active session>

Please let me know if you need more detail.   This appears to be new behaviour since next-3.1

Thanks

**Comment by nicm** [comment]
Show logs please.

**Comment by systemmonkey42** [comment]
Client and server logs.

https://www.dropbox.com/s/fay55rbv4s40fto/tmux-out-18178.log?dl=0

https://www.dropbox.com/s/32hcb6p68guqyfv/tmux-server-18178.log?dl=0

https://www.dropbox.com/s/qoe4qe5j0lvbw21/tmux-client-18176.log?dl=0

This was produced using

tmux -vvvv new
(in the first session)
$ tmux new -d

**Comment by nicm** [comment]
You have a script or something running switch-client which is probably what is changing your session:
~~~~
1574982818.322816 cmdq_fire_command </dev/pts/5>: (1463) new-session
1574982818.714262 cmdq_fire_command </dev/pts/9>: (1466) display-message -p "#{session_id}"
1574982818.721651 cmdq_fire_command </dev/pts/9>: (1469) switch-client -Z -t "$0"
1574982818.735343 cmdq_fire_command </dev/pts/9>: (1472) show-environment -s
1574982818.755355 cmdq_fire_command </dev/pts/9>: (1475) show-environment DISPLAY
1574982819.409012 cmdq_fire_command </dev/pts/9>: (1478) display-message -p "#{session_id}"
1574982819.413813 cmdq_fire_command </dev/pts/9>: (1481) switch-client -Z -t "$0"
1574982819.421981 cmdq_fire_command </dev/pts/9>: (1484) show-environment -s
1574982822.710555 cmdq_fire_command </dev/pts/9>: (1487) new-session -d
1574982823.129249 cmdq_fire_command </dev/pts/10>: (1490) display-message -p "#{session_id}"
1574982823.134626 cmdq_fire_command </dev/pts/10>: (1493) switch-client -Z -t "$1"
1574982823.143729 cmdq_fire_command </dev/pts/10>: (1496) show-environment -s
1574982823.160068 cmdq_fire_command </dev/pts/10>: (1499) show-environment DISPLAY
1574982823.725474 cmdq_fire_command </dev/pts/10>: (1502) display-message -p "#{session_id}"
1574982823.730130 cmdq_fire_command </dev/pts/10>: (1505) switch-client -Z -t "$1"
1574982823.737913 cmdq_fire_command </dev/pts/10>: (1508) show-environment -s
1574982827.893023 cmdq_fire_command </dev/pts/5>: (1511) attach-session

**Comment by systemmonkey42** [comment]
Ok, this is embarrasing, but you are absolutely right.

It turns out I was using 
```sess=$(tmux display-message -p "#{session_id}")```
followed by
```tmux switch-client -t "$sess"```
which did nothing previously, but had the side effect of updating environment.
I could then use:
```source  <(tmux show-environment -s)```
to update the environment.

eg, 

if I created two session client via SSH, and killed one of the off, the 'DISPLAY=' and 'SSH_AUTH_SOCK=' variables would now be invalid.  When I reconnected, the environment settings
in the TMUX server would be updated, and I could pull them into the active shell with the command ```source <(tmux show-environment -s)```

Already connected clients would not get the new variables, however the commands shown above would trigger and update of the environment, without actually switching the client.

This has changed in the newer release.  I found a work-around along the lines of
```source <$(tmux display-message -p "current=#{session_name} client=#{client_session")```
followed by
```[[ "${current}" == "${client}" ]] && switch-client -t "${current}"```

This allowed me to preserve the behaviour where updated DISPLAY and SSH variables were imported into the active client without accidentally making every session attempt to become the active session in the current client.

My apologies for wasting your time, and thanks for your great work.
